--- ResIL-1.8.3/m4/devil-definitions.m4.orig	2014-08-21 18:35:48.000000000 +0200
+++ ResIL-1.8.3/m4/devil-definitions.m4	2016-05-20 18:58:27.951271202 +0200
@@ -273,6 +273,16 @@
 			[enable_$1="yes"]) ])
 
 dnl Test for restrict c/c++ keyword
+AC_DEFUN([DEVIL_CHECK___RESTRICT],
+         [AC_MSG_CHECKING([__restrict keyword])
+	  AC_COMPILE_IFELSE(AC_LANG_SOURCE([void foo(void * __restrict test){}]),
+	            	    [AC_DEFINE([__RESTRICT_KEYWORD],
+				       [],
+				       [__restrict keyword available])
+                             restrict_keyword="found"
+	                     AC_MSG_RESULT([yes]) ],
+			    [AC_MSG_RESULT([no]) ]) ])
+
 AC_DEFUN([DEVIL_CHECK_RESTRICT],
          [AC_MSG_CHECKING([restrict keyword])
 	  AC_COMPILE_IFELSE(AC_LANG_SOURCE([void foo(void * restrict test){}]),
--- ResIL-1.8.3/include/IL/il_constants.h.orig	2014-11-15 16:01:26.000000000 +0100
+++ ResIL-1.8.3/include/IL/il_constants.h	2016-05-20 19:01:19.014597356 +0200
@@ -74,8 +74,8 @@
 #endif
 
 #ifdef RESTRICT_KEYWORD
-#define RESTRICT restrict
-#define CONST_RESTRICT const restrict
+#define RESTRICT __restrict
+#define CONST_RESTRICT const __restrict
 #else
 #define RESTRICT
 #define CONST_RESTRICT const
--- ResIL-1.8.3/configure.ac.orig	2016-05-20 16:32:31.754971998 +0200
+++ ResIL-1.8.3/configure.ac	2016-05-20 19:02:14.314595036 +0200
@@ -151,6 +151,8 @@
 
 dnl Test for restrict c/c++ keyword
 DEVIL_CHECK_RESTRICT
+AS_IF([test  "x$restrict_keyword" = "xfound"], [],
+      [DEVIL_CHECK___RESTRICT])
 
 dnl checking if the -std=gnu99 flag is required
 AS_IF([test  "x$restrict_keyword" = "xfound"], [],
